Feature/m0078 ddg 442 add development tools dependency discoverer (#407)
MicroEJ/docs: ApplicationDeveloperGuide/dependencyDiscoverer.rst

> .. _dependencyDiscoverer:

Reviewer: The build has failed https://readthedocs.com/projects/microej-docs/builds/957472/
Please check links or potential warnings

Author: passed https://readthedocs.com/projects/microej-docs/builds/958390/

=====================
Dependency Discoverer
=====================

Introduction
============

> 	The Dependency Discoverer is a tool designed to list missing dependencies of a program in the MicroEJ's central repository.

Reviewer: Please reuse the github readme introduction:

Dependency Discoverer is a tool that lists unresolved dependencies (types, methods and fields) of a set of Java ARchive (JAR) files and .class files.

Author: done in e3d0ebe0e67c3bc99a505d4c8883b1ada13edffb

> 	MicroEJ Dependency Discoverer is a tool that lists unresolved dependencies (types, methods and fields) of a set of Java ARchive (JAR) files and .class files.

Reviewer: ```suggestion
	Dependency Discoverer is a tool that lists unresolved dependencies (types, methods and fields) of a set of Java ARchive (JAR) files and .class files.
```

	It is a versatile tool and can be used in other contexts, for instance, to list every dependency of a JAR file.
	
	It can be used through a command-line interface, with the possibility to output the result in JSON or XML format, allowing an easy scripting process.
	

Installation
============

	This tool is available at https://github.com/MicroEJ/Tool-ApiDependencyDiscoverer. A JAR and Windows executable version can be downloaded from the release page.

> 	It is also possible to clone the project, open it into MicroEJ's SDK and use it from sources.

Reviewer: ```suggestion
	It is also possible to clone and import the project in the SDK and use it from sources.
```

Author: done in e3d0ebe0e67c3bc99a505d4c8883b1ada13edffb
